pyphare/pyphare/pharesee/hierarchy.py

```diff
@@ -149,9 +149,29 @@ def __init__(self, layout, field_name, data, **kwargs):
 
         self.dataset = data
 
+    def compare(self, that, atol=1e-16, rtol=0):
+        if isinstance(that, FieldData):
+            assert self.name == that.name
+            # drop nans
+            a = self.dataset
+            b = that.dataset
+            mask = ~(np.isnan(a) | np.isnan(b))
+            if __debug__:
+                try:
+                    np.testing.assert_allclose(a[mask], b[mask], atol=atol, rtol=rtol)
+                except AssertionError as e:
+                    print(f"FieldData comparison failure in {self.name}")
+                    raise e
+                return True
+            return np.allclose(a[mask], b[mask], atol=atol, rtol=rtol)
```

Comment:
having some report class could be an idea to be able to interrogate these things after the fact